src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -247,8 +246,11 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 // We don't need to genProduceReg(node) since that will be handled by processing op2
                 // likewise, processing op2 will ensure its own registers are consumed
 
-                // Make sure we consume the registers that are getting specially handled
-                genConsumeReg(op1);
+                if (!mergeWithZero)
+                {
+                    // Make sure we consume the registers that are getting specially handled
+                    genConsumeReg(op1);
```

Comment:
why we need a condition here? Doesn't `genConsumeReg()` become no-op if `op1` was contained?

Reply:
Not trivially, and there was an assert being fired. In general we don't want to consume a register if no register exists.